task/attest/src/main.rs

```diff
@@ -272,6 +274,54 @@ impl idl::InOrderAttestImpl for AttestServer {
         Ok(())
     }
 
+    fn reset(
+        &mut self,
+        msg: &userlib::RecvMessage,
+    ) -> Result<(), RequestError<AttestError>> {
+        if !PERMIT_LOG_RESET.iter().any(|x| *x == msg.sender.0) {
+            return Err(ClientError::AccessViolation.fail());
+        }
+
+        // Reset the attestation log
+        ringbuf_entry!(Trace::ResetLog);
+        self.measurements = Log::default();
+        Ok(())
+    }
+
+    fn reset_and_record(
+        &mut self,
+        msg: &userlib::RecvMessage,
+        algorithm: HashAlgorithm,
+        data: idol_runtime::Leased<idol_runtime::R, [u8]>,
+    ) -> Result<(), RequestError<AttestError>> {
+        ringbuf_entry!(Trace::ResetRecord(algorithm));
+
+        if !PERMIT_LOG_RESET.iter().any(|x| *x == msg.sender.0) {
+            return Err(ClientError::AccessViolation.fail());
+        }
+        ringbuf_entry!(Trace::ResetLog);
+        self.measurements = Log::default();
+
+        let measurement = match algorithm {
+            HashAlgorithm::Sha3_256 => {
```

Comment:
Should we instead make the `algorithm` argument into a data-bearing enum, e.g. `HashMeasurement::Sha3_256([u8; 32])`?

Reply:
I think that filhp was thinking about the various formats that other attestation schemes need to support and that we might want to support in the future. That doesn't mean it could not be done with a data-bearing enum. I'm not motivate to change this right now but can check with @flihp

Reply:
There will be changes to the attestation API when it is finally plumbed through. This PR won't try to anticipate those changes.